### DISCUSSION TITLE
How to enrich collector with host metrics?

### DISCUSSION THREAD

**Discussion by Jaans** [open-telemetry/opentelemetry-collector#4071] [Q&A] [answered]
Are we able to add some metrics (CPU, RAM, Storage, Network) at the OTEL collector level when collecting trace information?

I'm not sure whether this is a separate process that generates metrics (that the collector should receive), or whether - what in my mind seems like an ideal place - the collector itself can grab some host metrics and also send that on to the "backend".

What is the idea / best practice for this sort of thing?

**Comment by mx-psi** [comment, marked as answer]
Hey, not sure if this addresses your question but the Collector contrib distro has a [`hostmetrics` receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver#host-metrics-receiver) that fetches basic host metrics. You could add a metrics pipeline to send these to your preferred backend

**Reply by Jaans** [reply]
Ooo... that looks exactly like what I'm looking for where the collector is also the local agent. I'll have a look at trying to get that setup and test it out.
Thank you!
